assets/js/components/ToggleableCaption.js

```diff
@@ -17,6 +18,13 @@ module.exports = class ToggleableCaption {
       return;
     }
 
+    if (this.thresholdWidth != null) {
+      const conditionMediaQuery = `(max-width: ${this.thresholdWidth}px)`;
+      if (!this.window.matchMedia(conditionMediaQuery).matches) {
```

Comment:
we should listen for changes in viewport size so that we can check dynamically whether this media query is still/not satisfied. Same goes for Popup.js